File: LSP+ISP/C#_Final/MarsRoverKata/Commands/BackwardCommand.cs
namespace MarsRoverKata.Commands
{
    public class BackwardCommand : ICommand
    {
        private IMovable Movable { get; set; }

        public BackwardCommand(IMovable movable)
        {
            Movable = movable;
        }

        public bool Execute()
        {
            return Movable.MoveBackward();
        }
    }
}

File: LSP+ISP/Java_Final/MarsRoverKata/src/main/java/MarsRoverKata/Commands/BackwardCommand.java
package MarsRoverKata.Commands;

import MarsRoverKata.IMovable;

public class BackwardCommand implements ICommand {
    private IMovable movable;

    public IMovable getMovable() {
        return movable;
    }

    public void setRover(IMovable movable) {
        this.movable = movable;
    }

    public BackwardCommand(IMovable movable) {
        this.movable = movable;
    }

    public boolean execute() {
        return movable.moveBackward();
    }
}

